pkg/river/internal/value/value.go

```diff
@@ -67,8 +67,10 @@ func Object(m map[string]Value) Value {
 // Array creates an array from the given values. A copy of the vv slice is made
 // for producing the Value.
 func Array(vv ...Value) Value {
-	ty := reflect.ArrayOf(len(vv), goAny)
-	raw := reflect.New(ty).Elem()
+	// Arrays should be slices otherwise any reference to them gets copied by
+	// value into a new pointer.
+	arrayType := reflect.SliceOf(goAny)
+	raw := reflect.MakeSlice(arrayType, len(vv), len(vv))
```

Comment:
This was causing me a headache with error decorating. Each reference to the array was getting a different pointer, so we weren't able to match the failing value to a node in the AST.

Keeping the array as a slice prevents creating new copies of the array every time it's referenced, though it does lead to one more allocation.